### PR TITLE
merge podTemplates instead of overriding

### PR DESCRIPTION
**Pull request by pritidesai**
# Changes

The podTemplate can be specified for any given pipelineTask at the pipelineRun.spec.taskRunSpecs.

The podTemplate can also be specified for the entire pipeline at the pipelineRun.spec.podTemplate.

Before this commit, the podTemplates at the pipelineRun level were ignored if it was specified at the taskRunSpecs. This resulted in the settings specified at the pipelineRun level not included along with the pipelineTask specific configuration. For example, there could be a common setting at the pipelineRun level to set the User and Group for all the pipelineTasks and a pipelineTask can have its own diskType. Without this commit, the taskRun runs with a podTemplate with the specified diskType but no user or group configurations.

Fixes https://github.com/tektoncd/pipeline/issues/6846 

/kind bug

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Merge podTemplate specified in pipelineRun.spec.taskRunSpecs[].podTemplate along with pipelineRun.spec.podTemplate instead of only considering the one specified at the taskRunSpecs. 
```

### CODE REVIEW

**Comment by dibyom** [comment]
LGTM! 
cc @lbernick

**Comment by pritidesai** [comment]
Thanks @dibyom 👍

@tektoncd/core-maintainers please help review this 🙏  I will cherry pick this into previous LTS releases once this is merged, Thanks!

**Review by lbernick — APPROVED** [review]

**Review by jerop — APPROVED** [review]
/lgtm

**Comment by pritidesai** [comment]
Since we have identified this a bug, backporting this to all active LTS releases (v0.41, v0.44, v0.47).

/cherry-pick release-v0.47.x

**Pull request merged**
